ldap/servers/slapd/daemon.c

```diff
@@ -2845,16 +2846,23 @@ catch_signals()
 #endif /* HPUX */
 
 static int
-get_configured_connection_table_size(void)
+get_connection_table_size(void)
 {
-    int size = config_get_conntablesize();
+    int size = 0;
+    int resrvdesc = 0;
     int maxdesc = config_get_maxdescriptors();
 
-    /*
-     * Cap the table size at nsslapd-maxdescriptors.
-     */
-    if (maxdesc >= 0 && size > maxdesc) {
-        size = maxdesc;
+    /* Validate configured reserve descriptors */
+    validate_num_config_reservedescriptors();
+
+    resrvdesc = config_get_reservedescriptors();
+    if (maxdesc > resrvdesc) {
+         size = (maxdesc - resrvdesc);
+    }
```

Comment:
What if maxdesc <= resrvdesc ?

Reply:
> What if maxdesc <= resrvdesc ?

maxdesc is the max number of allowed open file descriptors on the system and resrvdesc is the number of reserve file descriptors for server use (SSL, backends, repl agmts, etc). IMHO maxdesc would never be > resrvdesc, but if you feel a check is required I can add one.

Reply:
That would means that there is a serious configuration issue that requires administrator action
  ( Probably need to  increase the system file descriptors limit, in this case 
      You may do a test log an error as fatal then exit